docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
-# Developer Guide
+# **Developer Guide**
 
-## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## **Introduction**
+ ![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/zoomwhitebg.png) <br/><br/>
+### **Welcome to Zoomaster!**
+Zoomaster is a Java application for the Command Line. It provides you with a simple and intuitive way to store your Zoom links for your online classes, alongside other useful links for the lesson at hand.<br/>
 
+Zoomaster can also intelligently determine the current lesson you are having, allowing you to launch the correct links quickly and elegantly from the command line.
 
-## Product scope
+
+### Table of contents
+* [Getting Started](#getting-started)
+* [Design](#design)
+  * [Architecture](#architecture)
+  * [UI component](#ui-component)
+  * [Logic component](#logic-component)
+  * [Model component](#model-component)
+  * [Storage component](#storage-component)
+  * [Common classes](#common-classes)
+* [Implementation](#implementation)
+  * [Bookmark and Timetable modes feature](#bookmark-and-timetable-modes-feature-tys)
+  * [Add Module and Slot feature](#add-module-and-slot-feature-xing-rong)
+* [Documentation, logging, testing, configuration, dev-ops](#architecture)
+* [Appendix: Requirements](#architecture)
+  * [Product Scope](#architecture)
+  * [User Stories](#architecture)
+  * [Use cases](#architecture)
+  * [Non-Functional Requirements](#architecture)
+  * [Glossary](#glossary)
+* [Appendix: Instructions for manual testing](#faq)
+  * [Launch and shutdown](#command-summary)
+
+
+## **Getting Started**
+Refer to the setting up guide over [here](https://github.com/AY2021S1-CS2113T-W11-1/tp/releases).
+
+
+
+
+
+
+## **Design**
+
+### Architecture
+
+The figure below shows a high-level design for the architecture of Zoomaster. 
+![](https://github.com/TYS0n1/tp/blob/team-Branch2/docs/diagrams/archiveture2-01.png?raw=true)
+*<center/> Figure 1.1 Architecture diagram of Zoomaster </center> <br/></br>*
+
+Our Program can be split up into 8 components
+* Initialization
+* User Interface
+* Parser
+* Commands
+* Temp Lists
+* Storage
+* Local Files
+* Exceptions
+
+These components interact with each other as shown in Figure 1.1 to execute the functionalities of Zoomaster.
+
+### Initialization
+The diagram below shows a class-level diagram for Zoomaster. <br/></br>
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/initial.png)
```

Comment:
Is it valid to put "use" label on the dash lines? Maybe it is non-standard notation used. 
Refer to this link for more details 
https://github.com/nus-cs2113-AY2021S1/forum/issues/112